src/app/pages/system/update/manual-update-form/manual-update-form.component.ts

```diff
@@ -0,0 +1,270 @@
+import { HttpErrorResponse } from '@angular/common/http';
+import { Component, Inject, OnInit } from '@angular/core';
+import { Validators } from '@angular/forms';
+import { MatDialog } from '@angular/material/dialog';
+import { Router } from '@angular/router';
+import { FormBuilder } from '@ngneat/reactive-forms';
+import { UntilDestroy, untilDestroyed } from '@ngneat/until-destroy';
+import { Store } from '@ngrx/store';
+import { TranslateService } from '@ngx-translate/core';
+import {
+  BehaviorSubject, noop, Observable, of, Subject,
+} from 'rxjs';
+import {
+  filter,
+  map, take, tap,
+} from 'rxjs/operators';
+import { JobState } from 'app/enums/job-state.enum';
+import { ProductType } from 'app/enums/product-type.enum';
+import { WINDOW } from 'app/helpers/window.helper';
+import { helptextSystemUpdate as helptext } from 'app/helptext/system/update';
+import { Job } from 'app/interfaces/job.interface';
+import { Option } from 'app/interfaces/option.interface';
+import { Preferences } from 'app/interfaces/preferences.interface';
+import { MessageService } from 'app/modules/entity/entity-form/services/message.service';
+import { EntityJobComponent } from 'app/modules/entity/entity-job/entity-job.component';
+import { EntityUtils } from 'app/modules/entity/utils';
+import { DialogService, SystemGeneralService, WebSocketService } from 'app/services';
+import { AppState } from 'app/store';
+import { waitForSystemInfo } from 'app/store/system-info/system-info.selectors';
+
+@UntilDestroy()
+@Component({
+  selector: 'manual-update-form',
+  templateUrl: './manual-update-form.component.html',
+  providers: [MessageService],
+  styleUrls: ['manual-update-form.component.scss'],
+})
+export class ManualUpdateFormComponent implements OnInit {
+  isFormLoading$: Subject<boolean> = new BehaviorSubject(false);
+  form = this.formBuilder.group({
+    filelocation: ['', Validators.required],
+    updateFile: [null as FileList],
+    rebootAfterManualUpdate: [false],
+  });
+  updateFile = '';
+  private get apiEndPoint(): string {
+    return '/_upload?auth_token=' + this.ws.token;
+  }
+
+  userPrefs: Preferences = null;
```

Comment:
You can use `selectPreferences` to get user preferences. To update them, you'd need to create an action in `preferences.actions.ts`.